src/parcsr_mv/par_csr_matrix.c

```diff
@@ -2246,13 +2197,13 @@ hypre_ParCSRMatrix * hypre_ParCSRMatrixUnion( hypre_ParCSRMatrix * A,
    hypre_ParCSRMatrixFirstRowIndex( C ) = hypre_ParCSRMatrixFirstRowIndex( A );
    hypre_assert( hypre_ParCSRMatrixFirstRowIndex( B )
                  == hypre_ParCSRMatrixFirstRowIndex( A ) );
-   hypre_ParCSRMatrixRowStarts( C ) = hypre_ParCSRMatrixRowStarts( A );
-   hypre_ParCSRMatrixOwnsRowStarts( C ) = 0;
-   hypre_ParCSRMatrixColStarts( C ) = hypre_ParCSRMatrixColStarts( A );
-   hypre_ParCSRMatrixOwnsColStarts( C ) = 0;
-   for ( p=0; p<=num_procs; ++p )
-      hypre_assert( hypre_ParCSRMatrixColStarts(A)
-                    == hypre_ParCSRMatrixColStarts(B) );
+   hypre_TMemcpy(hypre_ParCSRMatrixRowStarts(C), hypre_ParCSRMatrixRowStarts(A),
```

Comment:
Same question as above.

Reply:
Correct! Thanks for catching that. Moving to stack would solve this indirectly :)